server/fishtest/api.py

```diff
@@ -99,59 +97,140 @@ class ApiView(object):
     def __init__(self, request):
         self.request = request
 
-    def require_authentication(self):
-        token = self.request.userdb.authenticate(
-            self.get_username(), self.request.json_body["password"]
-        )
-        if "error" in token:
-            raise HTTPUnauthorized(token)
+    def validate_request(self, api):
+        error = ""
+        exception = HTTPBadRequest
+        for _ in range(1):  # trick to be able to use break
```

Comment:
Interesting trick! Perhaps better to skip the `range()` call using or `for _ in [None]:` or `while True:` (adding a last `break`). The latter IMO is less error prone because is clear that any exit condition needs a `break`

In this post a comparison of alternatives (IMO lesser alternatives for this case):
https://www.naftaliharris.com/blog/nontrivial-exception-handling-python/